web/src/features/public-api/components/ApiKeyList.tsx

```diff
@@ -22,26 +22,47 @@ import {
 import { usePostHogClientCapture } from "@/src/features/posthog-analytics/usePostHogClientCapture";
 import { CreateApiKeyButton } from "@/src/features/public-api/components/CreateApiKeyButton";
 import { useHasProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
+import { useHasOrganizationAccess } from "@/src/features/rbac/utils/checkOrganizationAccess";
 import { api } from "@/src/utils/api";
 import { DialogDescription } from "@radix-ui/react-dialog";
 import { TrashIcon } from "lucide-react";
 import { useState } from "react";
 import { Alert, AlertDescription, AlertTitle } from "@/src/components/ui/alert";
+import { startCase } from "lodash";
 
-export function ApiKeyList(props: { projectId: string }) {
-  const hasAccess = useHasProjectAccess({
-    projectId: props.projectId,
-    scope: "apiKeys:read",
+type ApiKeyScope = "project" | "organization";
+type ApiKeyEntity = { id: string; note: string | null };
+
+export function ApiKeyList(props: { entityId: string; scope: ApiKeyScope }) {
+  const { entityId, scope } = props;
+  if (!entityId) {
+    throw new Error(
+      `${scope}Id is required for ApiKeyList with scope ${scope}`,
+    );
+  }
+
+  const hasProjectAccess = useHasProjectAccess({
+    projectId: props.entityId,
+    scope: "apiKeys:CUD",
+  });
+  const hasOrganizationAccess = useHasOrganizationAccess({
+    organizationId: props.entityId,
+    scope: "organization:CRUD_apiKeys",
```

Comment:
@marcklingen Do we have any convention around scope names? I just created this one as it reflects best what we intend to cover with it. Initially, I thought `organizations:update` would also be sufficient, but having a separate scope here may make more sense.

Let me know if you have any doubts about this approach.